File: 13-roman-to-integer/roman-to-integer.cpp
class Solution {
public:
    int romanToInt(string s) {
        map<char,int> mpp;
       mpp['I']=1;
       mpp['V']=5;
       mpp['X']=10;
       mpp['L']=50;
       mpp['C']=100;
       mpp['D']=500;
       mpp['M']=1000;
       
        int result=0;
        for(int i=0;i<s.size();i++){
            if(mpp[s[i]]<mpp[s[i+1]])
            {
                result=result-mpp[s[i]];
            }
            else
            {
            result=result+mpp[s[i]];
            }
        }

        
       return result;
    }
};

File: 13-roman-to-integer/roman-to-integer.java
class Solution {
    public int romanToInt(String s) {
        Map<String,Integer>mpp=new HashMap<>();
        mpp.put("I",1);
        mpp.put("V",5);
        mpp.put("X",10);
        mpp.put("L",50);
        mpp.put("C",100);
        mpp.put("D",500);
        mpp.put("M",1000);
        int ans=0;
        for(int i=0;i<s.length()-1;i++){
            if(mpp.get(String.valueOf(s.charAt(i)))<mpp.get(String.valueOf(s.charAt(i+1)))){
                ans-=mpp.get(String.valueOf(s.charAt(i)));
            }
            else{
                ans+=mpp.get(String.valueOf(s.charAt(i)));
            }
        }
        ans+=mpp.get(String.valueOf(s.charAt(s.length()-1)));
        return ans;
    }
}